src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs

```diff
@@ -424,41 +424,7 @@ private IEnumerable<string> GetUnavailableMemberNames()
                 return false;
             }
 
-            private bool IsSymbolAccessible(
-                ISymbol symbol)
-            {
-                if (symbol == null)
-                {
-                    return false;
-                }
-
-                if (symbol.Kind == SymbolKind.Property)
-                {
-                    if (!IsSymbolAccessible(((IPropertySymbol)symbol).SetMethod))
-                    {
-                        return false;
-                    }
-                }
-
-                // Public and protected constructors are accessible.  Internal constructors are
-                // accessible if we have friend access.  We can't call the normal accessibility
-                // checkers since they will think that a protected constructor isn't accessible
-                // (since we don't have the destination type that would have access to them yet).
-                switch (symbol.DeclaredAccessibility)
-                {
-                    case Accessibility.ProtectedOrInternal:
-                    case Accessibility.Protected:
-                    case Accessibility.Public:
-                        return true;
-                    case Accessibility.ProtectedAndInternal:
-                    case Accessibility.Internal:
-                        return _document.SemanticModel.Compilation.Assembly.IsSameAssemblyOrHasFriendAccessTo(
-                            symbol.ContainingAssembly);
-
-                    default:
-                        return false;
-                }
-            }
+            private bool IsSymbolAccessibleToDocument(ISymbol symbol) => IsSymbolAccessible(symbol, _document);
```

Comment:
In an earlier version of this PR we did switch the IDE to use the new APIs. But that is no longer possible in the new shape of the compiler APIs. However, this little piece of the change remains: deleting duplicated code in the IDE by calling the duplicated code.

---
In reply to: [202476794](https://github.com/dotnet/roslyn/pull/27312#discussion_r202476794) [](ancestors = 202476794)

Reply:
> But that is no longer possible in the new shape of the compiler APIs.

Could you clarify why that is?

Note: if we go this path, i'm not sure if either of the referenced issues are actually addressed.  For example, https://github.com/dotnet/roslyn/issues/24518 says:

> See https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions_Accessibility.cs
> These language questions (of accessibility) should be implemented in the compiler layer, not the workspaces layer.

This PR won't actually address that.

note: this may be totally acceptable.  But it is worth noting that it effectively only sounds like it's addressing the concern about accessibility between symbols from the same compilation (maybe just the same language?).  This likely won't be sufficient for IDE scenarios and some subsets of users who want to work across languages.

Reply:
> Could you clarify why that is?

Yes. This API requires a compilation. It may be that there is a transformation locally at every place the Workspaces API is used, but doing that would now be a separate task.

> This PR won't actually address that.

Correct. But it provides a tool for addressing that. Users who want to work across languages or compilations can use this API after making a compilation that includes references to the assemblies containing the symbols of interest.